contracts/TransmuteDPOS.sol

```diff
@@ -112,14 +110,15 @@ contract TransmuteDPOS is TransmuteToken, RoundManager, ProviderPool {
     Provider storage p = providers[_provider];
     // A delegator is only allowed to bond to himself (in which case he wants to be a Provider)
     // or to a Registered Provider
-    require(_provider == msg.sender || p.status == ProviderStatus.Registered);
+    ProviderStatus pStatus = providerStatus(_provider);
+    require(_provider == msg.sender || pStatus == ProviderStatus.Registered);
```

Comment:
Why have a temp variable here and not just use the function in the equality check like you did on line `102`?